src/main/java/jenkins/bouncycastle/api/BCPasswordException.java

```diff
@@ -24,6 +24,9 @@
 
 package jenkins.bouncycastle.api;
 
+/**
+ * Indicates a password or passphrase is needed and none or a wrong one was provided
+ */
 public class BCPasswordException extends RuntimeException {
```

Comment:
http://ted-gao.blogspot.ie/2013/03/best-practices-in-java-exception.html

> ### Don’t Create Custom Exception Unnecessarily
> If possible, try to use exceptions in stand JDK, well-established libraries or framework, rather than create new custom exception. When a new custom exception is created, it takes time for other people to learn it. It also increases the size of our source code. Doing so only when it brings true benefits that existing exception classes cannot bring.